improver_tests/cli/test_init.py

```diff
@@ -197,111 +193,259 @@ def test_with_output(self, m):
         self.assertEqual(result, 4)
 
 
-def replace_load_with_extract(func, cube, constraints=None):
-    """Function to replicate the call to maybe_coerce_with within
-    create_constrained_inputcubelist_converter.
-
-    Args:
-        func:
-            Unused argument for the purpose of replicating the
-            maybe_coerce_with interface.
-        cube (iris.cube.Cube or iris.cube.CubeList):
-            Cube or CubeList to be extracted from.
-        constraints (str or None):
-            Expected name of cube for extraction.
-
-    Returns:
-        iris.cube.Cube:
-            The extracted cube.
-
-    Raises:
-        ValueError:
-            Error to replicate the behaviour of the call of maybe_coerce_with,
-            where loading a cube with an invalid constraint results in a
-            ValueError.
-    """
-    del func
-    if constraints:
-        cube = cube.extract(constraints)
-    if isinstance(cube, CubeList):
-        cube, = cube.copy()
-    if cube is None:
-        raise ValueError
-    return cube
+def list_of_length(length):
+    """Returns a list of the length plus 1"""
+    return list(range(length+1))
 
 
 class Test_create_constrained_inputcubelist_converter(unittest.TestCase):
     """Tests the creature constraint_inputcubelist_converter"""
 
     def setUp(self):
-        data = np.zeros((2, 2), dtype=np.float32)
-        self.wind_speed_cube = set_up_variable_cube(data, name="wind_speed")
-        self.wind_dir_cube = set_up_variable_cube(
-            data, name="wind_from_direction")
-        self.wind_cubes = CubeList([self.wind_speed_cube, self.wind_dir_cube])
-
-    @patch('improver.cli.maybe_coerce_with', return_value='return')
+        """Sets up some example names to use"""
+        self.speed_name = 'wind_speed'
+        self.direction_name = 'wind_from_direction'
+        # There is an early call to get the cubelist length
+        self.basic_coerse_calls = 1
+        self.fake_path = '/super/secret/data.nc'
+
+    @patch('improver.cli.maybe_coerce_with', side_effect=[list_of_length(2),
+                                                          'cube1', 'cube2'])
```

Comment:
So the effect of this statement is to change the return value of `maybe_coerce_with` into a list of (in this case) `[3, 'cube1', 'cube2']`?
I'm probably misinterpreting the documentation for Mock here, perhaps we should discuss.

Reply:
side effect will return a different item each time, so the first time it returns a list of length 3 `[1, 2, 3]` the second time it returns a `cube1` the third time `cube2`